ModularTegustation/tegu_items/workshop/_templates.dm

```diff
@@ -10,150 +10,68 @@
 	attack_verb_simple = list("poke", "jab", "tear", "lacerate", "gore")
 	hitsound = 'sound/weapons/ego/spear1.ogg'
 	var/active
-	var/list/finishedname = list()
-	var/list/finishedicon = list()
 	var/finisheddesc = "A finished weapon."
 	var/specialmod
-	var/aoe_range
+	var/aoe_range = 0
 	var/type_overriden = FALSE
 	var/forceholder	//holds the force for later
 	var/special_count //Various vars use this for various things
+	var/list/finishedname = list()
+	var/list/finishedicon = list()
 
 /obj/item/ego_weapon/template/attack(mob/living/target, mob/living/carbon/human/user)
 	forceholder = force
 	if(!active)
 		to_chat(user, "<span class='notice'>This weapon is unfinished!</span>")
 		return
-	specialcheck(target, user)
+	if(specialmod)
+		ModEffect(specialmod, target, user)
 	..()
 	if(forceholder != force)
 		force = forceholder
 
-/obj/item/ego_weapon/template/proc/specialcheck(mob/living/target, mob/living/carbon/human/user)
-	if(aoe_range)
-		aoe(target, user)
-	switch(pick(specialmod))
-		if("health healing")
-			HealHealth(target, user)
-		if("sanity healing")
-			HealSanity(target, user)
-		if("curing")
-			Cure(target, user)
-		if("split damage")
-			Split(target, user)
-		if("sharp")		//Uses Special Count for poise.
-			sharp(target, user)
-
 /obj/item/ego_weapon/template/attackby(obj/item/I, mob/living/user, params)
 	..()
-	if(!istype(I, /obj/item/workshop_mod) || active)
+	if(istype(I, /obj/item/workshop_mod) && !active)
+		InstallMod(I)
 		return
-	var/obj/item/workshop_mod/mod = I
+
+//I know its a proc containing one line of text but it simplifies the mod effect thing. -IP
+/obj/item/ego_weapon/template/proc/ModEffect(obj/item/workshop_mod/the_mod, mob/living/target, mob/living/carbon/human/user)
+	the_mod.ActivateEffect(src, special_count, target, user)
```

Comment:
This could be replaced by simply subtyping the variable `specialmod` to be of type `obj/item/workshop_mod` which would allow it to access the ActivateEffect proc immediately